### ISSUE TITLE
Access accuracy per user

### ISSUE THREAD

**Issue by giuliodavide** [massquantity/LibRecommender#83]
Hi to all,
I'm new to this library and I find it very interesting and useful. 
I would like to know if there's a way to access accuracy per user after the training part of a model.
Thank you in advance for your availability.

**Comment by massquantity** [comment]
```python
# assume `data` is a `pandas.DataFrame` which has format like this:
# user  item  label
#  1      2     1
#  2      3     0
#  1      1     0
#  3      7     1

import numpy as np
import pandas as pd
from pandas import DataFrame
from sklearn.metrics import accuracy_score

def compute_accuracy(df: DataFrame):
    # `predict` function returns probabilities, so one needs to convert them to label 1 or 0.
    probs = model.predict(df.user, df.item)
    preds = [1 if i >= 0.5 else 0 for i in probs]
    labels = df.label
    return accuracy_score(labels, preds)

model = ... 
accuracy_per_user = data.groupby("user").apply(compute_accuracy)
```

**Comment by giuliodavide** [comment]
Thank you very much.
By considering data a DataFrame containing the list of recommendation per user with respective labels it worked.

**Comment by giuliodavide** [comment]
I followed your suggestion but it continues to return me two exceptions: 
the first one is "'numpy.float64' object is not iterable", referred to probs which in one case returns only one probability (not a list);
the second one is "'DataFrame object has no attribute 'user'" even if I'm using a DataFrame as the one you told me in the first comment (user, item, label). I think I solved it by setting the attribute as_index=False in the groupby function, but it still returns me the former exception.
Other suggestions?

**Comment by massquantity** [comment]
Yep some models may return a single probability if there is only one sample for a user, so just add some checking:
```python
from typing import Iterable

def compute_accuracy(df: DataFrame):
    probs = model.predict(df.user, df.item)
    preds = (
        [1 if i >= 0.5 else 0 for i in probs]
        if isinstance(probs, Iterable)
        else [1 if probs >= 0.5 else 0]
    )
    labels = df.label
    return accuracy_score(labels, preds)
```
